File: jphp-javafx-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/UXTextInputControl.java
package org.develnext.jphp.ext.javafx.classes;

import javafx.scene.control.TextInputControl;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection.Abstract;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Property;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Abstract
@Name(JavaFXExtension.NAMESPACE + "UXTextInputControl")
public class UXTextInputControl extends UXControl {
    interface WrappedInterface {
        @Property int anchor();
        @Property int caretPosition();
        @Property int length();
        @Property String promptText();
        @Property String selectedText();
        @Property String text();

        @Property boolean editable();

        void appendText(String text);
        void backward();
        void clear();
        void copy();
        void cut();

        void deselect();

        void end();
        void endOfNextWord();

        void extendSelection(int pos);
        void forward();
        void home();

        void insertText(int index, String text);
        void replaceSelection(String replacement);
        void replaceText(int start, int end, String text);

        void paste();
        void positionCaret(int pos);

        void nextWord();
        void previousWord();
        void selectAll();
        void selectBackward();
        void selectEnd();
        void selectEndOfNextWord();
        void selectForward();
        void selectHome();
        void selectNextWord();
        void selectPositionCaret(int pos);
        void selectPreviousWord();
        void selectRange(int anchor, int caretPosition);
    }

    public UXTextInputControl(Environment env, TextInputControl wrappedObject) {
        super(env, wrappedObject);
    }

    public UXTextInputControl(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    public TextInputControl getWrappedObject() {
        return (TextInputControl) super.getWrappedObject();
    }
}

File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/UXTextInputControl.php
<?php
namespace php\gui;
use php\gui\text\UXFont;

/**
 * Class UXTextInputControl
 * @package php\gui
 * @packages gui, javafx
 */
abstract class UXTextInputControl extends UXControl
{
    /**
     * @var int
     */
    public $anchor;

    /**
     * @var string
     */
    public $text;

    /**
     * @var UXFont
     */
    public $font;

    /**
     * array(start, end, length).
     * @var array
     */
    public $selection;

    /**
     * @var string
     */
    public $selectedText;

    /**
     * @var string
     */
    public $promptText;

    /**
     * @readonly
     * @var int
     */
    public $length;

    /**
     * @var bool
     */
    public $editable;

    /**
     * @var int
     */
    public $caretPosition;

    public function copy() {}
    public function cut() {}
    public function paste() {}
    public function clear() {}

    public function end() {}
    public function home() {}
    public function forward() {}
    public function backward() {}
    public function nextWord() {}
    public function previousWord() {}

    public function selectAll() {}
    public function selectBackward() {}
    public function selectEnd() {}
    public function selectEndOfNextWord() {}
    public function selectForward() {}
    public function selectHome() {}
    public function selectNextWord() {}
    public function selectPreviousWord() {}

    /**
     * @param int $pos
     */
    public function selectPositionCaret($pos) {}

    /**
     * @param int $anchor
     * @param int $caretPosition
     */
    public function selectRange($anchor, $caretPosition) {}

    /**
     * @param int $pos
     */
    public function extendSelection($pos) {}

    /**
     * Deselect all.
     */
    public function deselect() {}

    /**
     * @param string $text
     */
    public function appendText($text) {}

    /**
     * @param int $index
     * @param string $text
     */
    public function insertText($index, $text) {}

    /**
     * @param int $start
     * @param int $end
     * @param string $text
     */
    public function replaceText($start, $end, $text) {}

    /**
     * @param string $text
     */
    public function replaceSelection($text) {}

    /**
     * @param int $pos
     */
    public function positionCaret($pos) {}

    /**
     * Undo changes.
     */
    public function undo()
    {
    }

    /**
     * Redo changes.
     */
    public function redo()
    {
    }

    /**
     * Commit the current text and convert it to a value.
     */
    public function commitValue()
    {
    }

    /**
     * If the field is currently being edited, this call will set text to the last commited value.
     */
    public function cancelEdit()
    {
    }
}